### DISCUSSION TITLE
Set thresholds not being applied for variables

### DISCUSSION THREAD

**Discussion by cwmac** [dtcenter/METplus#2747] [Configuration] [answered]
Trying to run verification for visibility and ceilings with thresholds under one mile and under 500 ft, respectively. Have set them in the config file and made sure the units match; however, the verification is only running for the whole dataset without any errors relating to the thresholds. 

I'm running METv11.1 on a Rocky Linux AWS box. Config file and log file from one of the runs is attached.

[thresholds_fail.txt](https://github.com/user-attachments/files/17514889/thresholds_fail.txt)
[GridStat_fcstHRRR_obsURMA_Sfc_MultiField.conf.SF.txt](https://github.com/user-attachments/files/17514892/GridStat_fcstHRRR_obsURMA_Sfc_MultiField.conf.SF.txt)

**Comment by j-opatz** [comment, marked as answer]
Hello,

And thank you for your question. After a review of your log files and the description of your problem, my coworker believes he's found the source of the issue as well as a fix (I'll use parts of his description for the following guidance).

You used `BOTH_VAR6_THRESH = "<=1609.34"` to set the threshold for `VIS` and `BOTH_VAR7_THRESH = "<=152.4"` to set the threshold for `HGT` in you METplus configuration file. However, you're requesting the CNT line type output. The `BOTH_VAR<n>_THRESH`setting sets the fcst.field.cat_thresh and obs.field.cat_thresh thresholds for MET which only impacts the computation of _CATEGORICAL_ statistics ( line types of CTC, CTS, etc.) and not the continuous statistics in the CNT line type. You can confirm this in the [METplus User's Guide section](https://metplus.readthedocs.io/en/latest/Users_Guide/wrappers.html#grid-stat-met-conf) that shows the MET wrapped configuration file for Grid-Stat and what each of the ${METPLUS_FCST_FIELD} and ${METPLUS_OBS_FIELD} settings control. 

To properly set the _continuous_ thresholds, you can utilize `BOTH_VAR6_OPTIONS = cnt_thresh = <=1609.34` for the `VIS` variable fields and `BOTH_VAR7_OPTIONS = cnt_thresh = <=152.4` for the `HGT` variable fields. 

We understand this issue may have been caused by confusion as to what the `BOTH_VAR<n>_THRESH` configuration was responsible for. We'll consider updating the configuration setting language in the future and we are currently in the process of adding direct support for CNT thresholds, which will remove the requirement for using _OPTIONS.

**Comment by cwmac** [comment]
That fixed the issue - thanks!
